### ISSUE TITLE
backend - node.js or ruby 

### ISSUE THREAD

**Issue by gg-1414** [tesla809/Pongzi#13]
best one to use??

**Comment by tesla809** [comment]
Depends. At some point we will be using Truffle and Ganache, which are development tools for Ethereum. I think they have only supported Node.js aka the javascript world. However, it would be fun to use if they have a Ruby gem for this. Can you check? If so, we can try it out?

What does the community thing @dinadeljanin @lookfwd @RANISio @mjohnson324

**Comment by tesla809** [comment]
https://truffleframework.com/
https://truffleframework.com/ganache

**Comment by lookfwd** [comment]
I'm sure Ruby has something, but the whole scene is ...

![image](https://user-images.githubusercontent.com/789359/48038396-672ca080-e13e-11e8-88ee-467fd6249a4b.png)

... so better to stick with the tools that have active developers that keep close track of the changes. JS has the additional advantage that you can more or less copy-paste stuff in the browser and it will work (more or less... and if web3.js happens to be about the same as the web3.js one uses with Truffle) :)

I think it's better to use the best supported tool for now. But if one wants to play with Ruby, why not?! :)

**Comment by mjohnson324** [comment]
Personally I agree with using tools with more support.

The company I'm at now is using Ruby on Rails, but truffle and ganache
handle everything related to ethereum.
You might check out these resources for Ruby-Blockchain interaction:

https://github.com/EthWorks/ethereum.rb (relatively new)
https://www.nopio.com/blog/blockchain-app-ruby/ (incorporating Rails)
https://github.com/Haseeb-Qureshi/lets-build-a-blockchain (tutorial on
making a blockchain in Ruby)
https://www.youtube.com/watch?v=3aJI1ABdjQk

The last two links I strongly recommend as a primer on blockchain and
crypto. It's by an App Academy instructor and he's good (I'm an App Academy
grad).

**Comment by lookfwd** [comment]
Yes, I agreeTo give some perspective, I love Python. I tried to use python with Ethereum for a few weeks. Most of my time was spent debugging web3py issues. Talking about down to the raw ETH JSON RPC API and hex messages. Huge waste. If I wanted to use this ecosystem in Python (or Ruby) I would write a wrapper with the most trivial JS ever, wrap my smart contract use cases in a REST API and then integrate with the rest of my infra.

**Comment by tesla809** [comment]
Great discussion guys: @lookfwd and @mjohnson324 ! Funny picture @lookfwd !
 @gg-1414 and @Frantz-B, What do you think? 

So, I am personally a Fullstack JS guy, but if the team wants to use Ruby/Python for some things, I would not mind. Always fun to learn new stuff. 

The question is where would we want to use Ruby or Python if not with dealing with Ethereum?
